shared-libs/couch-request/src/couch-request.js

```diff
@@ -0,0 +1,114 @@
+const request = require('request-promise-native');
+const isPlainObject = require('lodash/isPlainObject');
+const servername = new URL(process.env.COUCH_URL).hostname;
+
+
+const isString = value => typeof value === 'string' || value instanceof String;
+const isTrue = value => isString(value) ? value.toLowerCase() === 'true' : value === true;
+
+const addServername = isTrue(process.env.ADD_SERVERNAME_TO_HTTP_AGENT);
+const methods = {
+  GET: 'GET',
+  POST: 'POST',
+  DELETE: 'DELETE',
+  PUT: 'PUT',
+  HEAD: 'HEAD'
+};
+
+
+const mergeOptions = (target, source, exclusions = []) => {
+  for (const [key, value] of Object.entries(source)) {
+    if (Array.isArray(exclusions) && exclusions.includes(key)) {
+      return target;
+    }
+    target[key] = value; // locally, mutation is preferable to spreading as it doesn't
+    // make new objects in memory. Assuming this is a hot path.
+  }
+  return target;
+};
+
+// When proxying to HTTPS from HTTP (for example where an ingress does TLS termination in an SNI environment),
+// not including a 'servername' for a request to the HTTPS server (eg, def.org) produces the 
+// following error:
+// 
+// 'ERR_TLS_CERT_ALTNAME_INVALID'
+// "RequestError: Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames:
+//  Host: abc.com. is not in the cert's altnames: DNS:def.org"
+// 
+// The addition of 'servername' resolves this error. See docs for 'tls.connect(options[, callback])'
+//  (https://nodejs.org/api/tls.html): "Server name for the SNI (Server Name Indication) TLS extension  It is the
+//  name of the host being connected to, and must be a host name, and not an IP address.".
+// 
+
+const testReqInputs = (method, first, second = {}) => {
```

Comment:
@garethbowen Refactoring this into its own function was purely to please SonarQube. Personally think the code is more readable and less complex in a single function.